authentication/src/main/java/io/camunda/authentication/tenant/TenantAttributeHolder.java

```diff
@@ -7,41 +7,33 @@
  */
 package io.camunda.authentication.tenant;
 
-import static io.camunda.zeebe.protocol.record.value.TenantOwned.DEFAULT_TENANT_IDENTIFIER;
 import static org.springframework.web.context.request.RequestAttributes.SCOPE_REQUEST;
 
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Optional;
-import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.lang.NonNull;
 import org.springframework.web.context.request.RequestContextHolder;
 
+/**
+ * Provides static typed access to a thread-local request attribute that holds the list of
+ * authenticated tenant ids for the request.
+ */
 public final class TenantAttributeHolder {
   private static final String ATTRIBUTE_KEY = "io.camunda.zeebe.gateway.rest.tenantIds";
 
   private TenantAttributeHolder() {}
 
-  public static List<String> tenantIds() {
-    final var requestAttributes = getCurrentRequestAttributes();
-    return getTenantsOrDefaultTenant(requestAttributes);
-  }
-
-  public static void withTenantIds(final List<String> tenantIds) {
-    final var requestAttributes = getCurrentRequestAttributes();
-    final var tenantsOptional = Optional.ofNullable(tenantIds);
-    requestAttributes.setAttribute(ATTRIBUTE_KEY, tenantsOptional, SCOPE_REQUEST);
+  public static List<String> getTenantIds() {
+    final Object tenantAttributeValue =
+        RequestContextHolder.currentRequestAttributes().getAttribute(ATTRIBUTE_KEY, SCOPE_REQUEST);
+    return (List<String>) tenantAttributeValue;
   }
 
-  private static RequestAttributes getCurrentRequestAttributes() {
-    return RequestContextHolder.currentRequestAttributes();
-  }
-
-  private static List<String> getTenantsOrDefaultTenant(final RequestAttributes requestAttributes) {
-    final var tenants = requestAttributes.getAttribute(ATTRIBUTE_KEY, SCOPE_REQUEST);
-
-    if (tenants != null) {
-      return ((Optional<List<String>>) tenants).orElse(null);
-    } else {
-      return List.of(DEFAULT_TENANT_IDENTIFIER);
-    }
+  public static void setTenantIds(@NonNull final Collection<String> tenantIds) {
+    Preconditions.checkNotNull(tenantIds);
+    final var requestAttributes = RequestContextHolder.currentRequestAttributes();
+    requestAttributes.setAttribute(ATTRIBUTE_KEY, ImmutableList.copyOf(tenantIds), SCOPE_REQUEST);
```

Comment:
It's not quite the same because `unmodifiableList()` requires a list and `copyOf()` takes any collection, but I can replace it with `.stream().toList()`.